server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -394,16 +395,17 @@ public String executor() {
                 initializeExtensionResponseHandler
             );
             inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
-        } catch (CompletionException e) {
-            if (e.getCause() instanceof TimeoutException) {
+        } catch (CompletionException | ConnectTransportException e) {
+            if (e.getCause() instanceof TimeoutException || e instanceof ConnectTransportException) {
```

Comment:
```suggestion
            if (e.getCause() instanceof TimeoutException || e.getCause() instanceof ConnectTransportException) {
```

Reply:
The e.getCause() is only necessary for the CompletionException, because the actual exception is wrapped by the CompletionException.  The logic won't work correctly if the exception is an instance of ConnectTransportException.